watchers/configmap_watcher_test.go

```diff
@@ -0,0 +1,153 @@
+		configMapMock = watchers.NewMockConfigMapInterface(mockCtrl)
+		watcherMock = watchers.NewMockInterface(mockCtrl)
+		resultCh = make(chan watch.Event)
+		logger = zap.New()
+		exitCallsCount = 0
+
+		configMapMock.EXPECT().Watch(gomock.Any(), gomock.Any()).Return(watcherMock, nil).AnyTimes()
+		watcherMock.EXPECT().ResultChan().Return(resultCh).AnyTimes()
+		watcherMock.EXPECT().Stop().AnyTimes()
+	})
+
+	It("do nothing when initialFieldValue is not modified following an Add event", func() {
+		go func() {
+			defer GinkgoRecover()
+			watchers.WatchForChanges(configMapMock, configMapName, configMapField, initialFieldValue, logger, initialResourceVersion, func() { exitCallsCount++ })
+		}()
+
+		to := corev1.ConfigMap{
+			Data: map[string]string{
+				configMapField: initialFieldValue,
+			},
+			ObjectMeta: metav1.ObjectMeta{
+				ResourceVersion: initialResourceVersion,
+			},
+		}
+
+		resultCh <- watch.Event{
+			Type:   watch.Added,
+			Object: &to,
+		}
+
+		// wait a bit
+		<-time.After(2 * time.Second)
+
+		Expect(exitCallsCount).To(Equal(0))
+	})
+
+	It("exit when value changed after Add event", func() {
+		go func() {
+			defer GinkgoRecover()
+			watchers.WatchForChanges(configMapMock, configMapName, configMapField, initialFieldValue, logger, initialResourceVersion, func() { exitCallsCount++ })
+		}()
+
+		to := corev1.ConfigMap{
+			Data: map[string]string{
+				configMapField: "othervalue",
+			},
+			ObjectMeta: metav1.ObjectMeta{
+				ResourceVersion: initialResourceVersion,
+			},
+		}
+
+		resultCh <- watch.Event{
+			Type:   watch.Added,
+			Object: &to,
+		}
+
+		// wait a bit
+		<-time.After(2 * time.Second)
+
+		Expect(exitCallsCount).To(Equal(1))
+	})
+
+	It("exit when the value changed in the configMap", func() {
+		go func() {
+			defer GinkgoRecover()
+			watchers.WatchForChanges(configMapMock, configMapName, configMapField, initialFieldValue, logger, initialResourceVersion, func() { exitCallsCount++ })
+		}()
+
+		to := corev1.ConfigMap{
+			Data: map[string]string{
+				configMapField: "anothervalue",
+			},
+			ObjectMeta: metav1.ObjectMeta{
+				ResourceVersion: initialResourceVersion,
+			},
+		}
+
+		resultCh <- watch.Event{
+			Type:   watch.Modified,
+			Object: &to,
+		}
+
+		// wait a bit
+		<-time.After(2 * time.Second)
+
+		Expect(exitCallsCount).To(Equal(1))
+	})
+
+	It("do nothing after Delete event", func() {
+		go func() {
+			defer GinkgoRecover()
+			watchers.WatchForChanges(configMapMock, configMapName, configMapField, initialFieldValue, logger, initialResourceVersion, func() { exitCallsCount++ })
+		}()
+
+		to := corev1.ConfigMap{
+			Data: map[string]string{
+				configMapField: initialFieldValue,
+			},
+			ObjectMeta: metav1.ObjectMeta{
+				ResourceVersion: initialResourceVersion,
+			},
+		}
+
+		resultCh <- watch.Event{
+			Type:   watch.Deleted,
+			Object: &to,
+		}
+
+		// wait a bit
+		<-time.After(2 * time.Second)
+
+		Expect(exitCallsCount).To(Equal(0))
+	})
+
+	AfterEach(func() {
```

Comment:
we use to have BeforeEach and AfterEach just together it's just easy to read/follow